lib/segment/src/data_types/order_by.rs

```diff
@@ -17,20 +18,31 @@ pub enum Direction {
 }
 
 impl Direction {
-    pub fn as_range(&self, from: FloatPayloadType) -> Range<FloatPayloadType> {
+    pub fn as_range_from<T>(&self, from: T) -> Range<T> {
         match self {
             Direction::Asc => Range {
                 gte: Some(from),
-                ..Default::default()
+                gt: None,
+                lte: None,
+                lt: None,
             },
             Direction::Desc => Range {
                 lte: Some(from),
-                ..Default::default()
+                gt: None,
+                gte: None,
+                lt: None,
             },
         }
     }
 }
 
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Clone)]
+#[serde(untagged)]
+pub enum StartFrom {
+    Float(FloatPayloadType),
+    Datetime(DateTime<Utc>),
```

Comment:
We might need `deserialize_with` here, so it could work with requests like:

```
POST collections/test_collection/points/scroll
{
  "limit": 2,
  "order_by": {
    "direction": "desc",
    "key": "date",
    "start_from": "2024-10-11 12:34"
  }
}
```

See https://github.com/qdrant/qdrant/pull/3549